### PR TITLE
adds check for loopback url in array

### LINKED ISSUE #336: ensure loopback url is in whitelisturls when required
We should create a task that checks for the loopback url on tasks that require it.

### PR DESCRIPTION
**Pull request by katiedavis**
### WHY are these changes introduced?

Fixes #336 

This ensures that before we authenticate, the loopback url is written to the partner dash.

### WHAT is this pull request doing?
Adds a task to check for loopback url in redirectwhitelisturl array

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,30 +37,6 @@ Developers should have some prior knowledge of the Shopify app ecosystem. Curren
...
-
-For Rails Apps:
-```
-https://<LIVE_NGROK_URL>/auth/shopify/callback
-http://app-cli-loopback.shopifyapps.com:3456
-```
-
-The first URL handles the authentication flow for adding your app to a Shopify development store. The second is currently necessary to allow your local development server to make Shopify API requests on behalf of your development app.
-
-> NOTE: A future release will handle this URL whitelisting process automatically. During the beta, it must be done through the Partner Dashboard.
```
Comment (tylerball) [inline README.md]: 😄

File: lib/shopify-cli/tasks/ensure_loopback_url.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module ShopifyCli
+  module Tasks
+    class EnsureLoopbackURL < ShopifyCli::Task
+      def call(ctx)
+        @ctx = ctx
+        api_key = Project.current.env.api_key
+        result = Helpers::PartnersAPI.query(ctx, 'get_app_urls', apiKey: api_key)
+        loopback = OAuth::REDIRECT_HOST.to_s
```
Comment (tylerball) [inline lib/shopify-cli/tasks/ensure_loopback_url.rb]: No `.to_s`, it's already a string

File: lib/shopify-cli/tasks/ensure_loopback_url.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        app = result['data']['app']
+        urls = app['redirectUrlWhitelist']
+        if urls.grep(/#{loopback}/).empty?
+          with_loopback = urls.push(loopback.to_s)
+          ShopifyCli::Helpers::PartnersAPI.query(@ctx, 'update_whitelisturls', input: {
+            redirectUrlWhitelist: with_loopback, apiKey: api_key
+          })
+        else
```
Comment (tylerball) [inline lib/shopify-cli/tasks/ensure_loopback_url.rb]: this clause is redundant

Comment (katiedavis) [reply lib/shopify-cli/tasks/ensure_loopback_url.rb]: I'm 'verbose' 🙃

File: README.md [hunk 4]
```diff
@@ -37,30 +37,6 @@ Developers should have some prior knowledge of the Shopify app ecosystem. Curren
 
 - If you don’t have one, [create a Shopify partner account](https://partners.shopify.com/signup).
 - If you don’t have one, [create a Development store](https://help.shopify.com/en/partners/dashboard/development-stores#create-a-development-store) where you can install and test your app.
-- In the Partner dashboard, [create a new app](https://help.shopify.com/en/api/tools/partner-dashboard/your-apps#create-a-new-app). You’ll need this app’s API credentials during the setup process. You can use any placeholder App URL when creating an app, then update it when your local development app is running.
```
Comment (tanema) [inline README.md]: 🎉

File: lib/shopify-cli/commands/authenticate.rb [hunk 5]
```diff
@@ -16,10 +17,6 @@ def call(args, _name)
             spinner.update_title("Authetication token stored")
           rescue OAuth::Error
             @ctx.puts("{{error:Failed to Authenticate}}")
-            if command == 'shop' || command.nil?
```
Comment (tanema) [inline lib/shopify-cli/commands/authenticate.rb]: 🎈

**Pull request merged**